kubevirt_api/tests/test_unit.py

```diff
@@ -0,0 +1,222 @@
+
+    aggregator.assert_metric("kubevirt_api.vmi.count", value=0)
+
+
+def test_emits_zero_can_connect_when_service_is_down(dd_run_check, aggregator, instance):
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = []
+    check.kube_client.get_vmis.return_value = []
+
+    with pytest.raises(Exception):
+        dd_run_check(check)
+
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=0,
+        tags=HEALTHZ_TAGS,
+    )
+
+
+def test_emits_one_can_connect_when_service_is_up(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = GET_VMS_RESPONSE["items"]
+    check.kube_client.get_vmis.return_value = GET_VMIS_RESPONSE["items"]
+
+    dd_run_check(check)
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=1,
+        tags=HEALTHZ_TAGS,
+    )
+
+
+def test_raise_exception_when_metrics_endpoint_is_bad(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [BAD_METRICS_HOSTNAME_INSTANCE])
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = GET_VMS_RESPONSE["items"]
+    check.kube_client.get_vmis.return_value = GET_VMIS_RESPONSE["items"]
+
+    with pytest.raises(Exception):
+        dd_run_check(check)
+
+    aggregator.assert_metric(
+        "kubevirt_api.can_connect",
+        value=1,
+        tags=HEALTHZ_TAGS,
+    )
+
+
+def test_raise_exception_cannot_connect_to_kubernetes_api(dd_run_check, aggregator, instance, mocker, caplog):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [instance])
+    with pytest.raises(
+        Exception,
+    ):
+        dd_run_check(check)
+
+    assert "Cannot connect to Kubernetes API:" in caplog.text
+
+
+def test_log_warning_healthz_endpoint_not_provided(dd_run_check, aggregator, instance, mocker, caplog):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+
+    new_instance = deepcopy(instance)
+    new_instance.pop("kubevirt_api_healthz_endpoint")
+
+    check = KubeVirtApiCheck("kubevirt_api", {}, [new_instance])
+
+    check._setup_kube_client = lambda: None
+    check.kube_client = MagicMock()
+    check.kube_client.get_vms.return_value = GET_VMS_RESPONSE["items"]
+    check.kube_client.get_vmis.return_value = GET_VMIS_RESPONSE["items"]
+
+    dd_run_check(check)
+
+    assert (
+        "Skipping health check. Please provide a `kubevirt_api_healthz_endpoint` to ensure the health of the KubeVirt API."  # noqa: E501
```

Comment:
nit- we don't need the noqa, we can split this into two lines

Reply:
same as above